Filter

heartjigi opened the pull request:
Add Filter feature

Commit:
add filter api

Commit:
add filter on user list

Commit:
Merge branch 'master' of https://github.com/sv-bootcamp/bridgeme-client into filter

Commit:
fix dropdown problem

Commit:
Merge branch 'master' of https://github.com/sv-bootcamp/bridgeme-client into filter

Commit:
add sublabel on Checkbox module

Commit:
make loading view on login

Commit:
save before pull

Commit:
pull from master

Commit:
edit filter

Commit:
save before pull

Commit:
Merge branch 'master' of https://github.com/sv-bootcamp/bridgeme-client into filter

Commit:
save

Commit:
add filter save

Commit:
add removing filter sequence on asyncstorage at sign out

solve unlimited loading on log in

SarahhhC commented on src/components/MyPage.js:
```diff
@@ -110,7 +110,8 @@ class MyPage extends Component {
     try {
       UserUtil.signOut((result, error) => {
-        AsyncStorage.removeItem('token', () => { Actions.login(); });
+        let keys = ['token', 'filter'];
```
Reset filter when sign out? Why?
Didn't back-end side save user filter information and adjust it?

heartjigi replied:
yes, actually we didn't save user's filter data.
So I saved the filter data on local. Let's talk about this issue @KyuaKwon @Stacyyya

SarahhhC commented on src/components/SignUp/CareerInfo.js:
```diff
@@ -115,9 +117,16 @@ class CareerInfo extends Component {
       }
     }
 
+    for (let i = 0; i < this.state.pressed.length; i++)
```
add { }

heartjigi replied:
solved

SarahhhC commented on src/components/SignUp/CareerInfo.js:
```diff
@@ -115,9 +117,16 @@ class CareerInfo extends Component {
 
+  onPress() {
+    for (let i = 0; i < this.state.pressed.length; i++)
+      this.state.pressed[i] = true;
```
Can you explain what this is? `state.pressed` for

heartjigi replied:
The purpose of this state val is activate just one dropdown.

SarahhhC commented on src/components/SignUp/CareerInfo.js:
```diff
@@ -153,21 +162,23 @@ class CareerInfo extends Component {
-                    onSelect={this.onSelect.bind(this)}>
-                    {this.getOptionSet(idx)}
-                  </Select>
+                <View key={idx} style={[styles.questionContainer, { zIndex: zIdx }]}>
```
check indentation

heartjigi replied:
solved

SarahhhC commented on src/components/Main.js:
```diff
@@ -181,43 +182,45 @@ class Main extends Component {
-        <ChannelList tabLabel="ios-chatbubbles" style={styles.tabView} me={this.props.me} />
-        <MyPage tabLabel="md-contact" me={this.props.me} />
-      </ScrollableTabView>
+          <ScrollableTabView
```
check indentation.

heartjigi replied:
solved

SarahhhC commented on src/components/SignUp/SignUpMETA.js:
```diff
@@ -128,21 +128,32 @@ export const CareerData = {
     'Career change',
     'Networking',
-    'Soft skills (e.g., Communication..)',
+    'Soft skills',
```
Why options and option filter's contents are different? No relation with them?

heartjigi replied:
In filter, we decide delete after e.g. , however, I send data with original string.

SarahhhC replied:
ok good!

SarahhhC commented on src/components/SignUp/SignUpMETA.js:
```diff
@@ -128,21 +128,32 @@ export const CareerData = {
-    ['Feelers', 'Thinkers'],
-    ['Judgers', 'Perceivers'],
+    ['Feeler', 'Thinker'],
+    ['Judger', 'Perceiver'],
```
Personalities changed? Do database need update? I think yes...
Please tell me about these things 😄

heartjigi replied:
Yeah, include this. we have to update datas, It was changed with Stacy

SarahhhC replied:
Ok, let me know about them!

SarahhhC commented on src/components/UserList/UserList.js:
```diff
@@ -37,13 +39,13 @@ class UserList extends Component {
     if (error) {
       alert(error);
     } else if (result) {
+
```
delete line and where is else part?

heartjigi replied:
this line for comments line (JSCS),
and this part is not related me...

SarahhhC replied:
Ok!

SarahhhC commented on src/utils/ApiUtil.js:
```diff
@@ -87,12 +87,13 @@ class ApiUtil extends Component {
   }
 
   getResponse(response) {
+    console.log(response);
```
delete console.

heartjigi replied:
solved

SarahhhC commented on src/utils/ApiUtil.js:
```diff
@@ -87,12 +87,13 @@ class ApiUtil extends Component {
     .catch((error) => {
-      if (JSON.stringify(error) !== '{}') callback(null, JSON.stringify(error));
-      else callback(null, error);
+      if (JSON.stringify(error) === '{}') callback(null, error);
```
let's add braces. { }

Commit:
apply comments

Commit:
apply comments

ace0625 commented on ios/Bridgeme.xcodeproj/project.pbxproj:
```diff
@@ -52,6 +52,7 @@
 		D260393C1DF7D9D0004DC6C9 /* SF-UI-Text-Light.otf in Resources */ = {isa = PBXBuildFile; fileRef = D26039371DF7D8E0004DC6C9 /* SF-UI-Text-Light.otf */; };
```
No need to upload this file in this PR

ace0625 commented on src/components/UserList/UserList.js:
```diff
@@ -58,8 +60,8 @@ class UserList extends Component {
 
-  componentDidMount() {
-    UserUtil.getMentorList(this.onServerCallback.bind(this));
+  componentWillReceiveProps(props) {
```
why did you delete componentDidMount? I believe it should be there.

ace0625 commented on src/utils/CheckBox.js:
```diff
@@ -20,15 +20,20 @@ class CheckBox extends Component {
 
   getIcon() {
     if (this.props.checked)
```
use curly brackets for if statements

heartjigi replied:
solved

ace0625 commented on src/utils/CheckBox.js:
```diff
@@ -20,15 +20,20 @@ class CheckBox extends Component {
     let backColor = this.props.checked  ? '#fafafa' : '#ffffff';
-
-    const { iconStyle, labelStyle } = this.props;
+    const {
```
It would be better set this const part out of render function.

heartjigi replied:
actually this style is from react native dropdown module.
if I change this style, there are many jobs.
Do you think this work is needed?

Commit:
change category of mentor list api userutil -> matchutil
apply filter on re run

Commit:
del project.pbxproj

Commit:
Merge branch 'master' of https://github.com/sv-bootcamp/bridgeme-client into filter

Commit:
buildConfiguration = "Release"

Commit:
Apply coments

Commit:
apply comments

Commit:
Merge branch 'master' of https://github.com/sv-bootcamp/bridgeme-client into filter

Commit:
pull from master

Commit:
Merge branch 'master' of https://github.com/sv-bootcamp/bridgeme-client into filter

# Conflicts:
#	ios/Bridgeme.xcodeproj/project.pbxproj

ifndefdeadmau5 commented:
LGTM and I'd like to suggest you write commit message more meaningfull and also it would be good if there is a single change at one commit. or at least you can explain about your pull request at the content. Thanks!

[![Approved with PullApprove](https://img.shields.io/badge/code--review-approved-brightgreen.svg)](https://pullapprove.com/sv-bootcamp/bridgeme-client/pull-request/133/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=sv-bootcamp/bridgeme-client)

Commit:
Change filter to main flow
Actions.main -> Actions.pop
then solved problems with chat.

add Alert when no result

Commit:
Move refresh factor after check filter valodation

heartjigi commented:
I try to write commit message more meaningfull!
Thanks!
